packages/hub/services/validators/card-space.ts

```diff
@@ -22,57 +54,132 @@ export default class CardSpaceValidator {
+    if (cardSpace.profileDescription && cardSpace.profileDescription.length > MAX_LONG_FIELD_LENGTH) {
       errors.profileDescription.push(`Max length is ${MAX_LONG_FIELD_LENGTH}`);
     }
 
-    if (cardSpace.profileName?.length > MAX_SHORT_FIELD_LENGTH) {
+    if (cardSpace.profileName && cardSpace.profileName.length > MAX_SHORT_FIELD_LENGTH) {
       errors.profileName.push(`Max length is ${MAX_SHORT_FIELD_LENGTH}`);
     }
 
-    if (cardSpace.profileCategory?.length > MAX_SHORT_FIELD_LENGTH) {
+    if (cardSpace.profileCategory && cardSpace.profileCategory.length > MAX_SHORT_FIELD_LENGTH) {
       errors.profileCategory.push(`Max length is ${MAX_SHORT_FIELD_LENGTH}`);
     }
 
-    try {
-      let urlObject = new URL(`https://${cardSpace.url}`);
+    // Validate URLs
+
+    let cardSpaceUrl = `https://${cardSpace.url}`;
+    let urlValid = validateUrl(cardSpaceUrl!).valid;
+
+    if (!urlValid) {
+      errors.url.push('Invalid URL');
+    } else {
+      let urlObject = new URL(cardSpaceUrl);
       if (!urlObject.hostname.endsWith('card.space')) {
         errors.url.push('Only card.space subdomains are allowed');
       }
-    } catch (error) {
-      errors.url.push('Invalid URL');
     }
 
-    if (cardSpace.url.split('.').length - 1 !== 2) {
+    if (cardSpace.url && cardSpace.url.split('.').length - 1 !== 2) {
       errors.url.push('Only first level subdomains are allowed');
     }
 
     let cardSpaceWithExistingUrl = (await this.cardSpaceQueries.query({ url: cardSpace.url }))[0];
 
-    if (cardSpaceWithExistingUrl) {
+    if (cardSpace.id !== cardSpaceWithExistingUrl?.id && cardSpaceWithExistingUrl) {
       errors.url.push('Already exists');
     }
 
+    if (!validateUrl(cardSpace.profileImageUrl!).valid) {
+      errors.profileImageUrl.push('Invalid URL');
+    }
+
+    if (!validateUrl(cardSpace.profileCoverImageUrl!).valid) {
+      errors.profileCoverImageUrl.push('Invalid URL');
+    }
+
+    // Validate text fields
+
+    if (cardSpace.bioTitle && cardSpace.bioTitle.length > MAX_SHORT_FIELD_LENGTH) {
+      errors.bioTitle.push(`Max length is ${MAX_SHORT_FIELD_LENGTH}`);
+    }
+
+    if (cardSpace.bioDescription && cardSpace.bioDescription.length > MAX_LONG_FIELD_LENGTH) {
+      errors.bioDescription.push(`Max length is ${MAX_LONG_FIELD_LENGTH}`);
+    }
+
+    if (cardSpace.donationTitle && cardSpace.donationTitle.length > MAX_SHORT_FIELD_LENGTH) {
+      errors.donationTitle.push(`Max length is ${MAX_SHORT_FIELD_LENGTH}`);
+    }
+
+    if (cardSpace.donationDescription && cardSpace.donationDescription.length > MAX_LONG_FIELD_LENGTH) {
+      errors.donationDescription.push(`Max length is ${MAX_LONG_FIELD_LENGTH}`);
+    }
+
+    // Validate links
+
+    if (cardSpace.links) {
+      cardSpace.links.forEach((linkItem, index) => {
+        let { title, url } = linkItem;
+
+        if (!title) {
+          errors.links.push(`Title must be present. Link index: ${index}`);
```

Comment:
It might be nice to have it addressable by the client vs having to be parsed out of the string. [This](https://gregsdennis.github.io/Manatee.Json/usage/pointer.html#an-overview-of-json-pointer) JSON Pointer overview suggests it would be `/data/attributes/2/title`